docs/design/gang-podgroup.md

```diff
@@ -0,0 +1,121 @@
+# Gang PodGroups
```

Comment:
Suggest to create a separate PR for design doc :)

Reply:
NP, it also conflicts with the newest master. I will rebase first.